service/share/discovery/peer_cache.go

```diff
@@ -0,0 +1,48 @@
+package discovery
+
+import (
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// PeerCache holds all discovered and available peers
+type PeerCache struct {
```

Comment:
I just realized there is a `peer.Set` that has exactly this.

Reply:
done

Reply:
I just noticed that `peer.Set` does not provide functionality to remove peer from set. So, I will add a custom set.